Add C implementation for `@stdlib/number/float64/base/signbit` (#554)
stdlib-js/stdlib: lib/node_modules/@stdlib/number/float64/base/signbit/README.md

<!-- /.examples -->

<!-- C interface documentation. -->  
  
* * *

<section class="c">

## C APIs

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

</section>

<!-- /.intro -->

<!-- C usage documentation. -->

<section class="usage">

### Usage

```c
#include "stdlib/number/float64/base/signbit.h"
```

> #### stdlib_base_float64_signbit( x, \*out )

Reviewer: ```suggestion
#### stdlib_base_float64_signbit( x )
```

Returns an integer indicating whether the sign bit for a double-precision floating-point number is on (`1`) or off (`0`).

```c
#include <stdint.h>

int8_t out;

> stdlib_base_float64_signbit( 3.14, &out );

Reviewer: ```suggestion
stdlib_base_float64_signbit( 3.14 );
```

```

The function accepts the following arguments:

-   **x**: `[in] double` input value.

```c
int8_t stdlib_base_float64_signbit( const double x );
```

</section>

<!-- /.usage -->

<!-- C API usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="notes">

</section>

<!-- /.notes -->

<!-- C API usage examples. -->

<section class="examples">

### Examples

```c

> #include "stdlib/number/float64/base/signbit.h"
> #include <stdint.h>
> #include <stdio.h>
> 
> int main() {
>     double x[] = { 3.14, -3.14, 0.0, -0.0, 4.0, 1.0, -1.0, 1.0e308, -1.0e308 };
> 
>     int8_t out;
>     int i;
>     for ( i = 0; i < 9; i++ ) {
>         stdlib_base_float64_signbit( x[ i ], &out );
>         printf( "%lf => signbit: %u\n", x[ i ], out );
>     }
> }

Reviewer: ```suggestion
#include "stdlib/number/float64/base/signbit.h"
#include <stdint.h>
#include <stdio.h>
#include <inttypes.h>

int main() {
    double x[] = { 3.14, -3.14, 0.0, -0.0, 4.0, 1.0, -1.0, 1.0e308, -1.0e308 };

    int8_t out;
    int i;
    for ( i = 0; i < 9; i++ ) {
        stdlib_base_float64_signbit( x[ i ], &out );
        printf( "%lf => signbit: %" PRId8 "\n", x[ i ], out );
    }
}
```

<!-- /.examples -->

<!-- C interface documentation. -->  
  
* * *

<section class="c">

## C APIs

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

</section>

<!-- /.intro -->

<!-- C usage documentation. -->

<section class="usage">

### Usage

```c
#include "stdlib/number/float64/base/signbit.h"
```

#### stdlib_base_float64_signbit( x, \*out )

> Extracts a boolean indicating if the sign bit for a double-precision floating-point number is on (true) or off (false).

Reviewer: ```suggestion
Returns an integer indicating whether the sign bit for a double-precision floating-point number is on (`1`) or off (`0`).
```

```c
#include <stdint.h>

> int8_t out;
> stdlib_base_float64_signbit( 3.14 );

Reviewer: ```suggestion
int8_t out = stdlib_base_float64_signbit( 3.14 );
```

```

The function accepts the following arguments:

-   **x**: `[in] double` input value.

> -   **out**: `[out] int8_t*` destination for unbiased signbit.

Reviewer: ```suggestion
```

```c

> void stdlib_base_float64_signbit( const double x, int8_t *out );

Reviewer: ```suggestion
int8_t stdlib_base_float64_signbit( const double x );
```

```

</section>

<!-- /.usage -->

<!-- C API usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="notes">

</section>

<!-- /.notes -->

<!-- C API usage examples. -->

<section class="examples">

### Examples

```c
#include "stdlib/number/float64/base/signbit.h"
#include <stdint.h>
#include <stdio.h>

int main() {
    double x[] = { 3.14, -3.14, 0.0, -0.0, 4.0, 1.0, -1.0, 1.0e308, -1.0e308 };

    int8_t out;
    int i;
    for ( i = 0; i < 9; i++ ) {
        stdlib_base_float64_signbit( x[ i ], &out );

>         printf( "%lf => signbit: %u\n", x[ i ], out );

Reviewer: Nit: need to use macro for printing `int8_t` values.

    }
}
```

</section>

<!-- /.examples -->

</section>

> <!-- /.examples -->

Reviewer: ```suggestion
<!-- /.c -->
```
